### DISCUSSION TITLE
(Help needed). lotus-local-net  release/v1.14.0   mult  node add

### DISCUSSION THREAD

**Discussion by alanmoleapfive** [filecoin-project/lotus#8516] [Storage Provider]
### Checklist

- [X] This is **not** a security-related bug/issue. If it is, please follow please follow the [security policy](https://github.com/filecoin-project/lotus/security/policy).
- [X] This is **not** a question or a support request. If you have any lotus related questions, please ask in the [lotus forum](https://github.com/filecoin-project/lotus/discussions).
- [X] This is **not** a new feature request. If it is, please file a [feature request](https://github.com/filecoin-project/lotus/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Ffeature&template=feature_request.yml) instead.
- [X] This is **not** an enhancement request. If it is, please file a [improvement suggestion](https://github.com/filecoin-project/lotus/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Fenhancement&template=enhancement.yml) instead.
- [X] I **have** searched on the [issue tracker](https://github.com/filecoin-project/lotus/issues) and the [lotus forum](https://github.com/filecoin-project/lotus/discussions), and there is no existing related issue or discussion.
- [X] I am running the [`Latest release`](https://github.com/filecoin-project/lotus/releases), or the most recent RC(release canadiate) for the upcoming release or the dev branch(master), or have an issue updating to any of these.
- [X] I did not make any code changes to lotus.

### Lotus component

- [ ] lotus daemon - chain sync
- [X] lotus miner - mining and block production
- [ ] lotus miner/worker - sealing
- [ ] lotus miner - proving(WindowPoSt)
- [ ] lotus miner/market - storage deal
- [ ] lotus miner/market - retrieval deal
- [ ] lotus miner/market - data transfer
- [ ] lotus client
- [ ] lotus JSON-RPC API
- [ ] lotus message management (mpool)
- [ ] Other

### Lotus Version

```text
lotus-local-net  release/v1.14.0
```

### Describe the Bug

In lotus-local-net release/v1.14.0
Starting a single node works fine
Start miner on another machine

Start the node from
lotus daemon --genesis=devgen.car --bootstrap=false
Connect to the genesis node
lotus net connect /ip4/192.168.1.47/tcp/1234/p2p/12D3KooWDtjKv2guoiuvAeEWMnNaBPP9DLCRfKMEy9Rdvay8wfNu
connect 12D3KooWDtjKv2guoiuvAeEWMnNaBPP9DLCRfKMEy9Rdvay8wfNu: success
View the synchronization status of the current node
lotus sync status
Create a BLS type wallet
$ lotus wallet new bls
t3r33j64poh7x6o3obvhdtqtz5ychmyrdnh2caawfflmfpcdezx23vljpetcq7zp3gifpkdpe6hbzazw3ojp6a
Install 100 FIL from the genesis node to the current wallet
$ lotus send t3r33j64poh7x6o3obvhdtqtz5ychmyrdnh2caawfflmfpcdezx23vljpetcq7zp3gifpkdpe6hbzazw3ojp6a 100
bafy2bzaceaayxqimfkaltimubnwgy4o4sr3wzzbxclfpaiu27zkx37iizmt6c
Check balance
$ lotus wallet balance
100 FIL
$ lotus wallet list
Address Balance Nonce Default
t3r33j64poh7x6o3obvhdtqtz5ychmyrdnh2caawfflmfpcdezx23vljpetcq7zp3gifpkdpe6hbzazw3ojp6a 100 FIL 0 X
It's normal here

Initialize a new node
When the new wallet has a balance, the miner can be initialized. If there is no balance, it cannot be initialized.
lotus-miner init --sector-size=2KiB --nosync
Waiting for a problem
2022-02-22T07:58:18.618Z INFO badger v2@v2.2007.2/levels.go:183 All 0 tables opened in 0s
2022-02-22T07:58:19.155Z INFO main lotus-miner/init.go:748 Pushed CreateMiner message: bafy2bzacebpcmr54hoz7ug6zo7e5okx6r44xhs7medbo2hdx4pzvyqd4t6bno
2022-02-22T07:58:19.155Z INFO main lotus-miner/init.go:749 Waiting for confirmation

Attempt to transfer money on the new node
lotus send t3ryxdpzcjtkrtrgavguwzw65euo7euzdjeggehbwnwtf4a35ysugjzrrbnatytm645qusyd66jctp7ghh2l5a 10
bafy2bzacebw2hlhtf3k2x7ftft6w3gsouyi2konqgbdip7ocxgtnrdpibnato
But the master node did not receive it. The problem should be that none of the messages from the new node were sent to the master node.

### Logging Information

```text
2022-02-22T07:58:19.155Z        INFO    main    lotus-miner/init.go:748 Pushed CreateMiner message: bafy2bzacebpcmr54hoz7ug6zo7e5okx6r44xhs7medbo2hdx4pzvyqd4t6bno
2022-02-22T07:58:19.155Z        INFO    main    lotus-miner/init.go:749 Waiting for confirmation
```

### Repo Steps

1. Run '...'
2. Do '...'
3. See error '...'
...

**Comment by alanmoleapfive** [comment]
1. Import data to the node

# create a simple file
echo "Hi my name is $USER" > hello.txt
#Import Data
lotus client import ./hello.txt
Import 1645511212821164372, Root bafkqad2ineqg26janzqw2zjanfzsacq
# View all <Data CID>s of the current node
lotus client local
1645511212821164372: bafkqad2ineqg26janzqw2zjanfzsacq @/home/lotus/hello.txt (import)



2. Check the miner order price

lotus client query-ask t01000
Ask: t01000
Price per GiB: 0.0000000005 FIL
Verified Price per GiB: 0.00000000005 FIL
Max Piece size: 2 KiB
Min Piece size: 256 B
 

2. Initiate a storage transaction

lotus client deal <Data CID> <miner> <price> <duration>

lotus client deal bafkqad2ineqg26janzqw2zjanfzsacq t01000 0.0000000005 518400


miner miner id
price
duration indicates how long the miner will keep your data, expressed in blocks, one block represents 45 seconds, one day 246060/45=1920
Return Deal CID after successful command

bafyreideob3ihiah56g7eq7tqhozsucy2an5ygnr5pt4fq5fbneyqelzze


3. Check the transaction status


lotus client list-deals

DealCid DealId Provider State State On Chain? Slashed? PieceCID Size Price Price Duration Verified
...eyqelzze 0 t01000 StorageDealClientFunding N , N , ...anvufkiq 127 B 0.000260458 FIL 520916 false false

4. Find according to the data cid
lotus client find <Data CID>

lotus client find bafkqad2ineqg26janzqw2zjanfzsacq

5. Retrieve data according to cid
lotus client retrieve <Data CID> <outfile>

lotus client retrieve bafkqad2ineqg26janzqw2zjanfzsacq hello2.txt
If outfile does not exist, it will be created in the lotus warehouse directory
This command will initialize the retrieval transaction and download the data to your computer, this process takes about 2 to 10 minutes






question:

lotus client list-deals

root@ba3f11e94b07:/home/lotus# lotus client list-deals

root@ba3f11e94b07:/home/lotus# lotus client retrieve bafkqad2ineqg26janzqw2zjanfzsacq hello2.txt
Failed to find file
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1b43d92]

goroutine 1 [running]:
github.com/filecoin-project/lotus/cli.glob..func33(0xc0004325c0, 0x0, 0x0)
/mnt/lotus-local-net/cli/client_retr.go:321 +0x4d2
github.com/urfave/cli/v2.(*Command).Run(0x5a2a9a0, 0xc000432280, 0x0, 0x0)
/root/go/pkg/mod/github.com/urfave/cli/v2@v2.2.0/command.go:164 +0x4dd
github.com/urfave/cli/v2.(*App).RunAsSubcommand(0xc000001b00, 0xc0004ddf00, 0x0, 0x0)
/root/go/pkg/mod/github.com/urfave/cli/v2@v2.2.0/app.go:427 +0xa49
github.com/urfave/cli/v2.(*Command).startApp(0x5a2ae20, 0xc0004ddf00, 0x7ffd62b1a874, 0x6)
/root/go/pkg/mod/github.com/urfave/cli/v2@v2.2.0/command.go:279 +0x6bb
github.com/urfave/cli/v2.(*Command).Run(0x5a2ae20, 0xc0004ddf00, 0x0, 0x0)
/root/go/pkg/mod/github.com/urfave/cli/v2@v2.2.0/command.go:94 +0x9cd
github.com/urfave/cli/v2.(*App).RunContext(0xc000001980, 0x444b7d0, 0xc000054050, 0xc00004e0a0, 0x5, 0x5, 0x0, 0x0)
/root/go/pkg/mod/github.com/urfave/cli/v2@v2.2.0/app.go:306 +0x810
github.com/urfave/cli/v2.(*App).Run(...)
/root/go/pkg/mod/github.com/urfave/cli/v2@v2.2.0/app.go:215
github.com/filecoin-project/lotus/cli.RunApp(0xc000001980)
/mnt/lotus-local-net/cli/helper.go:35 +0x7c
main.main()
/mnt/lotus-local-net/cmd/lotus/main.go:109 +0x811
root@ba3f11e94b07:/home/lotus# lotus client find bafkqad2ineqg26janzqw2zjanfzsac
ERROR: length greater than remaining number of bytes in buffer

**Comment by Reiers** [comment]
Hello again @alanmoleapfive 

Thanks for the report. The issue you are experiencing lies elsewhere.

Moving this issue to Lotus Discussions for help and troubleshooting.

Closing Lotus issue ticket.

**Comment by LexLuthr** [comment]
@alanmoleapfive Can you please clarify the original description.

1. How are you connecting these nodes?
2. Are they on the same physical machines?
3. What is the use case for this scenario?
4. What is new node where you are attempting to transfer the money?
5. Please convert the description into a proper example with names and other identifiers for clarification.

**Comment by LexLuthr** [comment]
I am archiving this discussion as there has been no new input from the author for more than 7 days.
